### PR TITLE
Frontend Updates

### PR DESCRIPTION
**Pull request by DarkPurple141**
## Summary
* Added Semester badges
* Cleaned up a bit more inconsistent formatting, misc updates
* Added an edit/settings button
* Added support for numReviews in backend

### Desktop

screen shot 2018-12-10 at 9 32 39 pm

### Mobile sizing

screen shot 2018-12-10 at 9 33 11 pm

### Normal Full Size Cards

screen shot 2018-12-10 at 9 34 14 pm

### CODE REVIEW

File: frontend/src/components/Interactions/Edit.vue [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<template>
+    <button
+        class="material-icons edit-button"
+        type="menu"
+        title="Toggle Menu"
+    >
+        more_vert
```
Comment (NunoDasNeves) [inline frontend/src/components/Interactions/Edit.vue]: Doesn't this make more sense as a menu which contains stuff like 'Edit', 'Delete' etc?

Comment (NunoDasNeves) [inline frontend/src/components/Interactions/Edit.vue]: Also, it jiggles when I click it for some reason

File: backend/data/degrees.js [hunk 2]
```diff
@@ -1066,12 +1066,6 @@ module.exports = [
       'musicology music technology composition music performance music creative practice',
     type: 'undergraduate'
   },
-  {
-    name: 'Placeholder (Do Not Delete)',
```
[the remark this answers is not in the document]
Comment (DarkPurple141) [reply backend/data/degrees.js]: ikr

File: frontend/src/components/Course/FeedCard.vue [hunk 3]
```diff
@@ -2,19 +2,31 @@
...
         <time>{{ published }}</time>
       </p>
-      <Category :recommend="recommend" v-if="cardType === 'Review'">
-        <p><b>{{ positiveOrNegativeText }}</b></p>
-      </Category>
-      <p v-else>
-        <span v-if="numAnswers === 0">Know the answer to this question?</span>
-        <span v-else>{{ numAnswers }} Answer{{ numAnswers > 1 ? 's' : '' }}</span>
-      </p>
+      <div class="categories">
```
[the remark this answers is not in the document]
Comment (DarkPurple141) [reply frontend/src/components/Course/FeedCard.vue]: Yeah I dunno. I know what you mean -- I played around with a few different options. I think it's not too bad. Is your concern that it's not clear what recommended means in that context?

File: frontend/src/components/Interactions/Edit.vue (hunk 1 above)
Comment (DarkPurple141) [reply frontend/src/components/Interactions/Edit.vue]: I think more_vert is okay, but this file shouldn't be called edit.

Comment (DarkPurple141) [reply frontend/src/components/Interactions/Edit.vue]: Lel what.

File: frontend/src/components/Course/FeedCard.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (DarkPurple141) [reply frontend/src/components/Course/FeedCard.vue]: Main thing to amend with these cat badges is probably the courselist page.

File: frontend/src/components/Interactions/Edit.vue (hunk 1 above)
Comment (NunoDasNeves) [reply frontend/src/components/Interactions/Edit.vue]: yes

Comment (NunoDasNeves) [reply frontend/src/components/Interactions/Edit.vue]: * only on firefox :/ it moves a few pixels right on mousedown and moves back on mouseup. idk why lel. dont worry about it imo

**Pull request merged**